tests/test_grouping.py

```diff
@@ -0,0 +1,341 @@
+        p2_vec = np.array([0, 0, 0, 1, 1, 1, 0, 1])
+        p3_vec = np.array([0, 1, 1, 0, 0, 0, 1, 0])
+
+        assert (pauli_to_binary(p1_op, wire_map=wire_map) == p1_vec).all()
+        assert (pauli_to_binary(p2_op, wire_map=wire_map) == p2_vec).all()
+        assert (pauli_to_binary(p3_op, wire_map=wire_map) == p3_vec).all()
+        assert (pauli_to_binary(identity, wire_map=wire_map) == np.zeros(8)).all()
+
+    def test_binary_to_pauli_no_wire_map(self):
+        """Test conversion of Pauli in binary vector representation to operator form when no
+        `wire_map` is specified."""
+
+        p1_vec = np.array([1, 0, 1, 0, 0, 1])
+        p2_vec = np.array([1, 1, 1, 1, 1, 1])
+        p3_vec = np.array([1, 0, 1, 0, 1, 1])
+        zero_vec = np.zeros(6)
+
+        p1_op = PauliX(0) @ PauliY(2)
+        p2_op = PauliY(0) @ PauliY(1) @ PauliY(2)
+        p3_op = PauliX(0) @ PauliZ(1) @ PauliY(2)
+        identity = Identity(0)
+
+        assert are_identical_pauli_words(binary_to_pauli(p1_vec), p1_op)
+        assert are_identical_pauli_words(binary_to_pauli(p2_vec), p2_op)
+        assert are_identical_pauli_words(binary_to_pauli(p3_vec), p3_op)
+        assert are_identical_pauli_words(binary_to_pauli(zero_vec), identity)
+
+    def test_binary_to_pauli_with_wire_map(self):
+        """Test conversion of Pauli in binary vector representation to operator form when
+        `wire_map` is specified."""
+
+        p1_vec = np.array([1, 0, 1, 0, 0, 1])
+        p2_vec = np.array([1, 1, 1, 1, 1, 1])
+        p3_vec = np.array([1, 0, 1, 0, 1, 0])
+        zero_vec = np.zeros(6)
+
+        wire_map = {Wires("alice"): 0, Wires("bob"): 1, Wires("ancilla"): 2}
+
+        p1_op = PauliX("alice") @ PauliY("ancilla")
+        p2_op = PauliY("alice") @ PauliY("bob") @ PauliY("ancilla")
+        p3_op = PauliX("alice") @ PauliZ("bob") @ PauliX("ancilla")
+        identity = Identity("alice")
+
+        assert are_identical_pauli_words(binary_to_pauli(p1_vec, wire_map=wire_map), p1_op)
+        assert are_identical_pauli_words(binary_to_pauli(p2_vec, wire_map=wire_map), p2_op)
+        assert are_identical_pauli_words(binary_to_pauli(p3_vec, wire_map=wire_map), p3_op)
+        assert are_identical_pauli_words(binary_to_pauli(zero_vec, wire_map=wire_map), identity)
+
+    def test_convert_observables_to_binary_matrix(self):
+        """Test conversion of list of Pauli word operators to representation as a binary matrix."""
+
+        observables = [Identity(1), PauliX(1), PauliZ(0) @ PauliZ(1)]
+
+        binary_observables = np.array(
+            [[0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0], [0.0, 0.0, 1.0]]
+        )
+
+        assert (convert_observables_to_binary_matrix(observables) == binary_observables).all()
+
+    def test_is_qwc(self):
+        """Determining if two Pauli words are qubit-wise commuting."""
+
+        n_qubits = 2
+        p1_vec = pauli_to_binary(PauliX(0) @ PauliY(1), n_qubits)
```

Comment:
Would be worth adding 1-2 examples with specifically having `qml.Identity` in there.